src/Cardano/Address/Style/Shelley.hs

```diff
@@ -364,6 +380,71 @@ instance Internal.DelegationAddress Shelley where
               invariantNetworkTag 16 (networkTag @Shelley discrimination)
           expectedLength = 1 + 2*publicKeyHashSize
 
+newtype Word7 = Word7 Word8
```

Comment:
What about putting `Word7` & the various related function in a new module `Data.Word7` so that we keep this one clean and, can also have some nice unit tests referring just to the Data.Word7 part ?

Reply:
good idea - I will do it in separate PR